src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/ConfigurationBindingGeneratorTests.cs

```diff
@@ -250,9 +252,9 @@ public class MyClass2 { }
             Assert.Single(r);
 
             string generatedSource = string.Join('\n', r[0].SourceText.Lines.Select(x => x.ToString()));
-            Assert.Contains($"public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, global::Program.MyClass0 obj) => {{ }};", generatedSource);
-            Assert.Contains($"public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, global::Program.MyClass1 obj, global::System.Action<global::Microsoft.Extensions.Configuration.BinderOptions>? configureOptions) => {{ }};", generatedSource);
-            Assert.Contains($"public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, string key, global::Program.MyClass2 obj) => {{ }};", generatedSource);
+            Assert.Contains("public static void Bind(this IConfiguration configuration, object? obj) => BindCoreMain(configuration, obj, configureOptions: null);", generatedSource);
```

Comment:
Why is this changing from the strong type `MyClass0 obj` to `object? obj`?

Reply:
Due to https://github.com/dotnet/runtime/pull/90340#discussion_r1297436089. I thought we could generate one Bind method for each overload (3) rather than many typed methods. I'm reverting this approach because it breaks type checks for polymorphic objects, which is what's causing the PR failures.